src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java

```diff
@@ -146,6 +146,7 @@ public void openMainMenu(PlayerProfile profile, int page) {
         if (isSurvivalMode()) {
             profile.getGuideHistory().clear();
         }
+        profile.getGuideHistory().setMainMenuPage(page);
```

Comment:
The guide history should not be touched in non-survival mode.
Can also store the guide history as a variable here if we're being honest.
```suggestion
        if (isSurvivalMode()) {
            profile.getGuideHistory().clear();
            profile.getGuideHistory().setMainMenuPage(page);
        }
```